src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extensions.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo } from 'react';
+import {
+  QueryEditorExtension,
+  QueryEditorExtensionConfig,
+  QueryEditorExtensionDependencies,
+} from './query_editor_extension';
+
+interface QueryEditorExtensionsProps extends QueryEditorExtensionDependencies {
+  configMap?: Record<string, QueryEditorExtensionConfig>;
+  componentContainer: Element;
+  bannerContainer: Element;
+}
+
+const QueryEditorExtensions: React.FC<QueryEditorExtensionsProps> = React.memo((props) => {
+  const { configMap, componentContainer, bannerContainer, ...dependencies } = props;
+
+  const sortedConfigs = useMemo(() => {
+    if (!configMap || !Object.keys(configMap)) return [];
```

Comment:
```suggestion
    if ('[object Object]' !== Object.prototype.toString.call(configMap) || Object.keys(configMap).length === 0) return [];
```
This will need linting though.

Reply:
thanks good point on `.length === 0`, i missed that

is the object check necessary or conventional in OSD?